### DISCUSSION TITLE
table with multiple relations on same key

### DISCUSSION THREAD

**Discussion by ctrenks** [prisma/prisma#13410] [Q&A] [answered]
I have a table that holds decritions for multiple different types of items, this is based on an id, type and parent.  Works great for one but when i try to connect 2 or more I get an error about using the same id again.

```
model descriptions {
  id                   Int     @id @default(autoincrement()) 
  parent            Int     
  title                String?
  description     String?
  type                int
  game_review      games @relation(references: [id], fields: [parent])
  review                casinos @relation(references: [id], fields: [parent])
}
```
so in this case games are type 4, casinos are type 3

on the query I check for the type

```
    review: {
        where: {
          type: 3,
        },
      },
```
This works great without 2 relations in the descriptions table.

**Comment by nurul3101** [comment, marked as answer]
Hey @ctrenks 👋 
Could you try this descriptions model and check?
```prisma
model descriptions {
  id          Int     @id @default(autoincrement())
  parent      Int
  title       String?
  description String?
  type        Int
  game_review games   @relation(references: [id], fields: [parent], map: "game_review_fkey")
  review      casinos @relation(references: [id], fields: [parent])
}
```

**Reply by ctrenks** [reply]
I added map: "casino_review_fkey" as well on the review and this works, can you explain WHY this works?  Is it simply adding another name for the parent field so its unique?

Thanks!

**Comment by ctrenks** [comment]
here is a new issue, the schema works but the generate does not like it

```
PS C:\VS\afc-redux\master> npx prisma db push 
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "db.vegldlkkofrlncfkbrzv.supabase.co:5432"
Error: insert or update on table "casino_p_descriptions" violates foreign key constraint "description_casino_fk"
   0: sql_migration_connector::apply_migration::apply_migration
             at migration-engine\connectors\sql-migration-connector\src\apply_migration.rs:9
   1: migration_core::state::SchemaPush
             at migration-engine\core\src\state.rs:349
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
